esdt-safe/src/lib.rs

```diff
@@ -94,28 +103,22 @@ pub trait EsdtSafe {
     fn get_next_transaction_batch(&self) -> SCResult<EsdtSafeTxBatch<Self::BigUint>> {
         self.require_caller_owner()?;
 
-        if self.pending_transaction_address_nonce_list().is_empty() {
-            return Ok(EsdtSafeTxBatch {
-                batch_id: 0,
-                transactions: Vec::new()
-            })
+        let current_block_nonce = self.blockchain().get_block_nonce();
+        let first_tx_block_nonce = match self.get_first_pending_transaction_block_nonce() {
```

Comment:
this seems wrong. What we want is that every transaction which is chosen for the next batch is at least Min_BLOCKS old.

Reply:
Transactions in the pending list are in a FIFO order of their creation time, so if the first one is old enough, the rest are guaranteed to be as least as old as that one.

Reply:
^False. Will fix.

Reply:
Fixed.